homeassistant/components/brother/sensor.py

```diff
@@ -84,260 +86,277 @@
     ),
 }
 
-SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Brother entities from a config_entry."""
+    coordinator = hass.data[DOMAIN][DATA_CONFIG_ENTRY][entry.entry_id]
+
+    sensors = []
+
+    device_info = DeviceInfo(
+        configuration_url=f"http://{entry.data[CONF_HOST]}/",
+        identifiers={(DOMAIN, coordinator.data.serial)},
+        manufacturer=ATTR_MANUFACTURER,
+        model=coordinator.data.model,
+        name=coordinator.data.model,
+        sw_version=getattr(coordinator.data, "firmware", None),
+    )
+
+    for description in SENSOR_TYPES:
+        if description.key in coordinator.data:
+            sensors.append(
+                description.entity_class(coordinator, description, device_info)
+            )
+    async_add_entities(sensors, False)
+
+
+class BrotherPrinterSensor(CoordinatorEntity, SensorEntity):
+    """Define an Brother Printer sensor."""
+
+    def __init__(
+        self,
+        coordinator: BrotherDataUpdateCoordinator,
+        description: BrotherSensorEntityDescription,
+        device_info: DeviceInfo,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        self._attrs: dict[str, Any] = {}
+        self._attr_device_info = device_info
+        self._attr_name = f"{coordinator.data.model} {description.name}"
+        self._attr_unique_id = f"{coordinator.data.serial.lower()}_{description.key}"
+        self.entity_description = description
+
+    @property
+    def native_value(self) -> StateType | datetime:
+        """Return the state."""
+        return cast(
+            StateType, getattr(self.coordinator.data, self.entity_description.key)
```

Comment:
I don't think we need to cast to `StateType | datetime` type here because `datetime` as a state can only appear in the `BrotherPrinterUptimeSensor` class.